js/blogtini.js

```diff
@@ -151,22 +151,33 @@ jekyll (GitHub Pages) plugins:
 /* eslint-disable no-continue */
 import yml from 'https://esm.archive.org/js-yaml'
 import dayjs from 'https://esm.archive.org/dayjs'
-import showdown from 'https://esm.archive.org/showdown'
 import hljs from 'https://esm.archive.org/highlightjs'
 
 import { krsort } from 'https://av.prod.archive.org/js/util/strings.js'
 
 // adds header click actions, etc.
 // eslint-disable-next-line import/no-named-as-default
-import search_setup from './future-imperfect.js'
 import { markdown_to_html, summarize_markdown } from './text.js'
 
 
 // eslint-disable-next-line no-console
 const log = console.log.bind(console)
+import {
+  assertBaseUrlWithEndingSlash,
+  cleanUpInitialPayloadMarkup,
+  createBlogtiniStuffWrapper,
+} from './utils.mjs'
 
+const SITE_ROOT_BASE_URL = new URL(import.meta.url).searchParams.get("SITE_ROOT_BASE_URL")
+const PRODUCTION_SITE_ROOT_BASE_URL = new URL(import.meta.url).searchParams.get("PRODUCTION_SITE_ROOT_BASE_URL")
+const FILE_SLASH_SLASH_SLASH_SITE_ROOT_BASE_URL = /^file\:\//.test(SITE_ROOT_BASE_URL)
 
+assertBaseUrlWithEndingSlash(PRODUCTION_SITE_ROOT_BASE_URL)
+assertBaseUrlWithEndingSlash(SITE_ROOT_BASE_URL)
+
+const dayJsHelper = dayjs()
 const state = {
+  top_dir: SITE_ROOT_BASE_URL,
```

Comment:
This is suggested (`SITE_ROOT_BASE_URL`, maybe not use "`top_dir`") so that what's `/theme.js` can know what's the `baseUrl`, if it's `file:///` it'll be different than on an `http://` with a `/` or `/subfolder` like GitHub pages often lets us do.